packages/FlexGrid/Col/Col.jsx

```diff
@@ -5,42 +5,65 @@ import { Subscriber } from 'react-broadcast'
 import { Col as ReactFlexboxGridCol } from 'react-flexbox-grid'
 
 import safeRest from '../../../shared/utils/safeRest'
+import { deprecate } from '../../../shared/utils/warn'
 import styles from './Col.modules.scss'
 
-const removeProps = ({
-  xs,
-  sm,
-  md,
-  lg,
-  xl,
-  xsOffset,
-  smOffset,
-  mdOffset,
-  lgOffset,
-  xlOffset,
-  ...rest
-}) => safeRest(rest)
-
-const Col = ({ span, offset, children, ...rest }) => (
-  <Subscriber channel="flex-grid">
-    {gutter => (
-      <ReactFlexboxGridCol
-        {...removeProps(rest)}
-        xs={span || true}
-        xsOffset={offset}
-        className={gutter ? styles.padding : styles.gutterless}
-      >
-        {children}
-      </ReactFlexboxGridCol>
-    )}
-  </Subscriber>
-)
+const removeProps = ({ xsOffset, smOffset, mdOffset, lgOffset, xlOffset, ...rest }) =>
+  safeRest(rest)
+
+const Col = ({ span, offset, xs, sm, md, lg, xl, children, ...rest }) => {
+  if (span) {
+    deprecate(
+      'core-flex-grid',
+      'The span prop is deprecated due to the addition of the xs, sm, md, lg and xl responsive props. Please replace the span prop in your Col definition with the xs prop for identical functionality.'
+    )
+  }
+
+  return (
+    <Subscriber channel="flex-grid">
+      {gutter => (
+        <ReactFlexboxGridCol
+          {...removeProps(rest)}
+          xs={xs || span || true}
+          sm={sm || 0}
```

Comment:
`defaultProps` would be the preferred approach here instead of the `||` operator.

However, in this case I'm not sure a default value of `0` is what we want. I think `undefined` is probably more correct. In that the number of columns for those viewports has not been specified, which is different than specifying them to be 0.

Reply:
I tried this, and setting these to undefined makes the grid take on generic classes for each breakpoint. For example, if lg is set to undefined, it will take on a generic lg class, meaning that if someone just wanted to set the md prop and have it apply to all greater sizes, it would not work because the generic lg class is overriding the md class. I can look into this more and see if there's a workaround. Good call on the default props.